Source/Classes/PINRemoteImageDownloadQueue.m

```diff
@@ -106,7 +105,7 @@ - (void)scheduleDownloadsIfNeeded
             NSURLSessionDataTask *task = [queue firstObject];
             [queue removeObjectAtIndex:0];
             [task resume];
-            
+            [task PIN_updateStartTime];
```

Comment:
So sadly, this isn't when the system will guarantee starting the task :( It's when we ask it to, but the system can choose to delay starting it…

Reply:
So method swizzling will not work neither, since it will update the start time immediately when `resume` is called.